infoschema/builder.go

```diff
@@ -111,9 +111,16 @@ func (b *Builder) ApplyDiff(m *meta.Meta, diff *model.SchemaDiff) ([]int64, erro
 		if err != nil {
 			return nil, errors.Trace(err)
 		}
+		if diff.Type == model.ActionTruncateTablePartition {
+			tblIDs = []int64{}
+		}
 	}
 	if diff.AffectedOpts != nil {
 		for _, opt := range diff.AffectedOpts {
+			if diff.Type == model.ActionTruncateTablePartition {
```

Comment:
better add a comment for the follower readers.

Reply:
ok, done.